MobileCenter/MobileCenter/Internals/Model/MSCustomPropertiesLog.m

```diff
@@ -0,0 +1,93 @@
+#import "MSCustomPropertiesLog.h"
+
+static NSString *const kMSCustomProperties = @"custom_properties";
+static NSString *const kMSProperties = @"properties";
+static NSString *const kMSPropertyType = @"type";
+static NSString *const kMSPropertyName = @"name";
+static NSString *const kMSPropertyValue = @"value";
+static NSString *const kMSPropertyTypeClear = @"clear";
+static NSString *const kMSPropertyTypeBoolean = @"boolean";
+static NSString *const kMSPropertyTypeNumber = @"number";
+static NSString *const kMSPropertyTypeDateTime = @"date_time";
+static NSString *const kMSPropertyTypeString = @"string";
+
+@implementation MSCustomPropertiesLog
+
+@synthesize type = _type;
+@synthesize properties = _properties;
+
+- (instancetype)init {
+  self = [super init];
+  if( self ) {
+    self.type = kMSCustomProperties;
+  }
+  return self;
+}
+
+#pragma mark - MSSerializableObject
+
+- (NSMutableDictionary *)serializeToDictionary {
+  NSMutableDictionary *dict = [super serializeToDictionary];
+  
+  if (self.properties) {
+    NSMutableArray *propertiesArray = [NSMutableArray array];
+    for (NSString *key in self.properties) {
+      NSObject *value = [self.properties objectForKey:key];
+      NSMutableDictionary *property = [MSCustomPropertiesLog serializeProperty: value];
+      if (property) {
+        [property setObject:key forKey:kMSPropertyName];
+        [propertiesArray addObject:property];
+      }
+    }
+    dict[kMSProperties] = propertiesArray;
+  }  return dict;
+}
+
++ (NSMutableDictionary *)serializeProperty:(NSObject *)value {
+  static NSDateFormatter *dateFormatter = nil;
```

Comment:
I normally don't like this but imho in this case, it makes sense to have the dateFormatter `static`.